src/github.com/couchbase/sync_gateway/db/crud.go

```diff
@@ -464,7 +464,7 @@ func (db *Database) updateDoc(docid string, allowImport bool, callback func(*doc
 	var docSequence uint64
 	var unusedSequences []uint64
 
-	err := db.Bucket.WriteUpdate(key, 0, func(currentValue []byte) (raw []byte, writeOpts sgbucket.WriteOptions, err error) {
+	oldBodyBytes, err := db.Bucket.WriteUpdate(key, 0, func(currentValue []byte) (raw []byte, writeOpts sgbucket.WriteOptions, oldBody []byte, err error) {
```

Comment:
Do we actually need to return oldBodyBytes from WriteUpdate, or can we just define it in the scope of updateDoc and have the callback update?